specs/DesignDiscussions/Template.md

```diff
@@ -0,0 +1,418 @@
+# Overview
+## Templating
+![](assets/Template/img1.png)
+**Figure 1**
+
+Template allows users to define where data binding occurs.
+
+```json
+{
+    "type": "TextBlock",
+    "text": "{latestPrice} ",
+    "size": "ExtraLarge"
+}
+```
+
+Once binding is defined, adaptive cards will be updated by changing only the data.
+```json
+{
+  "latestPrice": 128.9,
+}
+```
+### Syntax
+- Binding syntax starts with { and ends with }. E.g., {myProperty}
+- Dot-notation to access sub-objects
+ - {employee.manager.name}
+- Indexer syntax to retrieve properties by key or items in an array
+ - {employee.peers[0].name}
+- $data assigns data context to current element
+- $when conditional expression that determines if current element to be included.
+- functions are provided to help SDK users
```

Comment:
Is this supposed to be a comprehensive syntax list of *all* well-known keywords in teh syntax? If not let's remove it, if yes then this should be comprehensive.